bin/patch

```diff
@@ -0,0 +1,139 @@
+#!/usr/bin/env perl
+# patch - Apply a patch to Webmin core or its modules from GitHub or a local file
+
+use strict;
+use warnings;
+
+use 5.010;
+
+use Getopt::Long qw(:config permute pass_through);
+use Pod::Usage;
+use Term::ANSIColor qw(:constants);
+use File::Basename;
+use Cwd qw(cwd);
+
+my %opt;
+GetOptions('help|h' => \$opt{'help'});
+pod2usage(0) if ($opt{'help'});
+
+# Get Webmin path
+my $path = cwd;
+my $lib = "web-lib-funcs.pl";
+if (!-r "$path/$lib") {
+	$path = dirname(dirname($0));
+	if (!-r "$path/$lib") {
+		$path = $path = Cwd::realpath('..');
+		}
+}
+
+# Check if curl is installed
+if (!`which curl`) {
+	print RED, "curl is not installed\n", RESET;
+	exit 1;
+	}
+
+# Check if git is installed
+if (!`which git`) {
```

Comment:
Also here

Reply:
Same here.

Reply:
I mean that I tried `has_command` in the first place and it didn't work. I assumed this is because of env.

Reply:
You should add some debugging code to check if $PATH is set

Reply:
Well, we could use `has_command` like this https://github.com/webmin/webmin/pull/2207/commits/d89f6411b62ae55e10e7bb48c87be64b1241aa78. But I think it's overhead and doesn't have any benefits?

Reply:
I think it's always better to use our standard APIs wherever possible. For example, that `which` command isn't always available on all operating systems and shells.

Reply:
Okay, is there anything else on this PR?